middle.mkd

```diff
@@ -274,8 +274,9 @@ A GeoJSON object with the type "Feature" is a feature object.
 A GeoJSON object with the type "FeatureCollection" is a feature
 collection object. A feature collection object MUST have a member with
 the name "features". The value of "features" is an array. Each element
-of the array is a feature object as defined above. It is possible for
-this array to be empty.
+of the array MUST be a feature object as defined above. It is possible
+for this array to be empty. The values in the array are ordered, meaning
+that GeoJSON implementations MUST maintain order when processing them.
```

Comment:
On 2016-02-18 00:28, Allan Doyle wrote:
> I content that we can talk about the GeoJSON object and fully and
> unambiguously describe it without reference to a processing model.

i don't think so, if you think about it as a data exchange format. to 
give you an example ad absurdum: if that were true, i could write 
GeoJSON apps that would depend on whitespace and all other kinds of JSON 
syntax details for semantics. i could generate GeoJSON where semantics 
depend on indentation. few people would be able to work with that, and 
few would agree (i guess) that this is helpful in terms of using GeoJSON 
as a meaningful exchange format.

by saying what matters, you make it clear what interop depends on. 
whitespace indentation on the JSON level probably is not one of those 
things. many other things probably also clearly would not be very 
helpful in using GeoJSON in an interoperable way.

in the data interchange formats i have worked with, a typical question 
always was "how do i expose this to users", in the sense of people 
writing libraries, and wanting to expose the format to library users. is 
that something you see as a goal as well? and if we don't have a way for 
a library to decide what to expose, it has no option other than exposing 
every last detail of the syntax, right?